tools/c7n_azure/c7n_azure/autoscale_utils.py

```diff
@@ -0,0 +1,55 @@
+import logging
+from c7n.utils import local_session
+from c7n_azure.session import Session
+from azure.mgmt.monitor import MonitorManagementClient
+
+
+class AutoScaleUtilities(object):
```

Comment:
I would suggest to move this to `provisioning` folder and inherit `deployment_unit` class.

It will give you some functionality for free (like SDK client initialization). We have some SDK client modifications around logging\error handling, so we try to avoid direct initialization for these objects.

Instead of doing manual validation I suggest to extend servicePlan schema (in `policy.py`) to enforce validation before the actual provisioning.